nextroute/schema/input.go

```diff
@@ -7,129 +7,217 @@ import (
-	MaxDuration             *int           `json:"max_duration,omitempty"`
-	MaxWait                 *int           `json:"max_wait,omitempty"`
-	ActivationPenalty       *int           `json:"activation_penalty,omitempty"`
-	StartLocation           *Location      `json:"start_location,omitempty"`
-	AlternateStops          *[]string      `json:"alternate_stops,omitempty"`
-	InitialStops            *[]InitialStop `json:"initial_stops,omitempty"`
-	ID                      string         `json:"id,omitempty"`
+	// Capacity of the vehicle.
+	Capacity any `json:"capacity,omitempty"`
+	// StartLevel initial level of the vehicle.
+	StartLevel any `json:"start_level,omitempty"`
+	// CustomData arbitrary custom data.
+	CustomData any `json:"custom_data,omitempty"`
+	// CompatibilityAttributes attributes that the vehicle is compatible with.
+	CompatibilityAttributes *[]string `json:"compatibility_attributes,omitempty"`
+	// MaxDistance maximum distance in meters that the vehicle can travel.
+	MaxDistance *int `json:"max_distance,omitempty"`
+	// StopDurationMultiplier multiplier for the duration of stops.
+	StopDurationMultiplier *float64 `json:"stop_duration_multiplier,omitempty"`
+	// StartTime time when the vehicle starts its route.
+	StartTime *time.Time `json:"start_time,omitempty"`
+	// EndTime latest time at which the vehicle ends its route.
+	EndTime *time.Time `json:"end_time,omitempty"`
+	// EndLocation location where the vehicle ends.
+	EndLocation *Location `json:"end_location,omitempty"`
+	// MinStops minimum stops that a vehicle should visit.
+	MinStops *int `json:"min_stops,omitempty"`
+	// MinStopsPenalty penalty for not visiting the minimum number of stops.
+	MinStopsPenalty *float64 `json:"min_stops_penalty,omitempty"`
+	// MaxStops maximum number of stops that the vehicle can visit.
+	MaxStops *int `json:"max_stops,omitempty"`
+	// Speed of the vehicle in meters per second.
+	Speed *float64 `json:"speed,omitempty"`
+	// MaxDuration maximum duration in seconds that the vehicle can travel.
+	MaxDuration *int `json:"max_duration,omitempty"`
+	// MaxWait maximum aggregated waiting time that the vehicle can wait across route stops.
+	MaxWait *int `json:"max_wait,omitempty"`
+	// ActivationPenalty penalty of using the vehicle.
+	ActivationPenalty *int `json:"activation_penalty,omitempty"`
+	// StartLocation location where the vehicle starts.
+	StartLocation *Location `json:"start_location,omitempty"`
+	// AlternateStops a set of alternate stops for which only one should be serviced.
+	AlternateStops *[]string `json:"alternate_stops,omitempty"`
+	// InitialStops initial stops planned on the vehicle.
+	InitialStops *[]InitialStop `json:"initial_stops,omitempty"`
+	// ID of the vehicle.
+	ID string `json:"id,omitempty"`
 }
 
 // InitialStop represents an initial stop.
 type InitialStop struct {
-	Fixed *bool  `json:"fixed,omitempty"`
-	ID    string `json:"id"`
+	// Fixed whether the stop is fixed or not.
```

Comment:
```suggestion
	// Fixed ensures that the stop stays with the vehicle it is defined on. Furthermore, the sequence of fixed stops will stay untouched. However, other stops may get assigned in between fixed stops.
```
We should explain what it does instead.